yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -199,8 +199,13 @@ class MapOperator : public ObservableOperator<U, D, MapOperator<U, D, F>> {
         : SuperSub(std::move(observable), std::move(observer)) {}
 
     void onNext(U value) override {
-      auto& map = SuperSub::getObservableOperator();
-      SuperSub::observerOnNext(map->function_(std::move(value)));
+      try {
```

Comment:
should we put this try..catch into observerOnNext?

Reply:
All `observerOnNext()` does is call `downstreamObserver_->onNext()`, which has nothing to do with the `F` function passed in `map(F)`.